### PR TITLE
Use Raiden Janitor instead of mirakuru

### PR DESCRIPTION
**Pull request by karlb**
Implements https://github.com/raiden-network/scenario-player/issues/468 in the most basic way. Followups to be discussed (feel free to add items during review):
- [ ] readd health check
- [ ] send SIGKILL to nodes that won't stop by themselves (the SIGKILL in https://github.com/raiden-network/raiden/pull/5974 might be enough)

### CODE REVIEW

File: scenario_player/runner.py [hunk 1]
```diff
@@ -371,7 +359,7 @@ def _initialize_udc(
         return ud_token_tx, udc_ctr, should_deposit_ud_token
 
     def _initialize_nodes(
-        self,
+        self, nursery: Any
```
Comment (palango) [inline scenario_player/runner.py]: Can't we type this properly?

File: scenario_player/runner.py [hunk 2]
```diff
@@ -404,7 +392,28 @@ def _initialize_nodes(
...
-        node_starter = self.node_controller.start(wait=False)
+        try:
+            for node_runner in self.node_controller._node_runners:
+                node_runner.start()
+        except Exception:
+            log.error("failed to start", exc_info=True)
+            raise
+
+        def wait():
+            with gevent.Timeout(100):
```
Comment (palango) [inline scenario_player/runner.py]: Maybe pull this into a constant for more visibility (I guess right now we're fast enough, but we could run into this)

File: scenario_player/runner.py [hunk 3]
```diff
@@ -276,34 +275,23 @@ def _setup(self, mint_gas, node_count, node_addresses, fund_tx, node_starter, gr
...
-        greenlets.add(root_task_greenlet)
-        try:
-            gevent.joinall(greenlets, raise_error=True)
-        except BaseException as exc:
-            log.exception("Greenlet Error", exception=exc)
-            if not root_task_greenlet.dead:
-                # Make sure we kill the tasks if a node dies
-                root_task_greenlet.kill()
-            raise
+        self.root_task()
```
Comment (palango) [inline scenario_player/runner.py]: Why don't we need to start this in a greenlet any more?

File: scenario_player/runner.py (hunk 1 above)
Comment (karlb) [reply scenario_player/runner.py]: The `Nursery` is defined inside the `Janitor` as a closure. I don't think mypy can access that. We can probably pull that out into a separate class with some additional constructor args. But that has to be done on the Janitor side first.

File: scenario_player/runner.py (hunk 3 above)
Comment (karlb) [reply scenario_player/runner.py]: This whole function is already run in a greenlet. I didn't see a need to add an additional greenlet. We could probably have removed that greenlet even without the other changes.

File: scenario_player/runner.py (hunk 1 above)
Comment (palango) [reply scenario_player/runner.py]: There's `class Nursery(ABC):` for that reason I think.

Comment (karlb) [reply scenario_player/runner.py]: Good point, I forgot about that!

**Pull request merged**